### ISSUE TITLE
Freely customizable WASM import

### ISSUE THREAD

**Issue by gguoss** [paritytech/substrate#1043]
https://github.com/paritytech/substrate/blob/d3c0b508b5bff4de7099aa1e2b504f183413ef07/core/executor/src/wasm_executor.rs#L144

I want to import the secp256k1 related function into WASM. 
However,
``` Rust
 struct FunctionExecutor<'e, E: Externalities<Blake2Hasher> + 'e> {
    sandbox_store: sandbox::Store,
    heap: Heap,
    memory: MemoryRef,
    table: Option<TableRef>,
    ext: &'e mut E,
    hash_lookup: HashMap<Vec<u8>, Vec<u8>>,
}
```
only use impl_function_executor!  to initialize function, Can't be customized.

**Comment by pepyakin** [comment]
@bkchr I noticed that you assigned yourself. Could you share your plan?

**Comment by bkchr** [comment]
@pepyakin Yes I did that, but it was by accident and I did remove myself directly again. So, I don't have any plan ^^
